File: app/src/main/java/com/ladsoft/bakingapp/mvp/model/TaskHandler.java
package com.ladsoft.bakingapp.mvp.model;

import android.os.Handler;
import android.os.HandlerThread;

public class TaskHandler extends HandlerThread {
    private static final String TAG = TaskHandler.class.getSimpleName();

    private Handler workerHandler;

    public TaskHandler() {
        super(TAG);
    }

    public void prepareWorkerHandler() {
        workerHandler = new Handler(getLooper());
    }

    public void postTask(Runnable task) {
        workerHandler.post(task);
    }
}

File: app/src/main/java/com/ladsoft/bakingapp/mvp/model/TaskHandler.kt
package com.ladsoft.bakingapp.mvp.model


import android.os.Handler
import android.os.HandlerThread

class TaskHandler : HandlerThread(TAG) {

    private var workerHandler: Handler? = null

    fun prepareWorkerHandler() {
        workerHandler = Handler(looper)
    }

    fun postTask(task: Runnable) {
        workerHandler?.post(task)
    }

    companion object {
        private val TAG = TaskHandler::class.java.simpleName
    }
}